xapian-core/weight/tfidfweight.cc

```diff
@@ -149,6 +149,10 @@ TfIdfWeight::get_idfn(Xapian::doccount termfreq, char c) const
 	    // All documents are indexed by the term
 	    if (N == termfreq) return 0;
 	    return log((N - termfreq) / termfreq);
+        case 'f':
+            return (1.0/termfreq);
+        case 's':
+            return pow(log(N/termfreq),2.0);
 	default:
```

Comment:
The indentation of these new lines isn't tab-filled (you can see in the github diff viewer that they don't quite line up with the old lines.  There's a handy script in the git repo `xapian-maintainer-tools/xapian-check-patch` which will check for this and other common issues.  You can feed it a patch/diff output, e.g. to check all changes between origin/master and your tree:

```
git diff origin/master | xapian-maintainer-tools/xapian-check-patch
```

There should also be spaces around `/` and after `,`.